src/common/types/function.ts

```diff
@@ -705,6 +715,11 @@ export class FunctionInstance {
 
                 if (item.type === 'Output') {
                     outputs.set(item.output.id, type);
+                    if (definition.schema.kind === 'regularNode' && inputLengths.size > 0) {
+                        // This only works because we assume sequences from a single node are all the same size.
+                        // In the future, this might not be the case.
+                        outputLengths.set(item.output.id, [...inputLengths.values()][0]);
+                    }
```

Comment:
The whole point of your recent iterator changes was to allow connecting 2 different iterators of a single node if they have the same length. You can't just assume they have the same length, in the system that is supposed to verify this very fact.